src/test/java/TestVector.java

```diff
@@ -0,0 +1,51 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TestVector {
+
+    Vector testVector1 = new Vector(1, 2, 3);
+    Vector testVector2 = new Vector(2, 2, 2);
+    Vector testVector3 = new Vector(7, 8, 9);
+
+    @Test
+    public void testVector() {
+
+        // check that constructor works well
+        assertEquals(8, testVector3.getCoordY(), "Not expected coordY in new vector");
+        testVector3.setCoordY(10);
+        assertEquals(10, testVector3.getCoordY(), "Not expected coordY in the vector after change");
+    }
+
+    @Test
+    public void testCalculateVectorLength() {
+        assertEquals(3.7416573867739413, testVector1.calculateVectorLength(), "incorrect vector's length");
+    }
+
+    @Test
+    public void testCalculateDotProduct() {
+        assertEquals(12.00d, Vector.calculateDotProduct(testVector1, testVector2), "incorrect dot product");
+    }
+
+    @Test
+    public void testGenerateVectorRandom() {
+
+        // Our random Vectors have coordinates in range 0<coordinate<100
+        // let`s check some of them
+        assertTrue(0.00d <= Vector.generateRandomVectors()[0].getCoordX() &&
+                Vector.generateRandomVectors()[0].getCoordY() <= 100.00d, "not expected coordY in new random vector");
+        assertTrue(0.00d <= Vector.generateRandomVectors()[1].getCoordY() &&
```

Comment:
please fix the assertions - this implementation will verify if y is bigger than 0 (and ignore that it can be more than 100) and that z is just less than 100 (but ignore that it can be less than 0 as well):
<img width="927" alt="image" src="https://user-images.githubusercontent.com/90613721/147005046-06df4a0a-aa22-48d4-9cf4-763b6935c4d7.png">

you can create simple method with that will take double as an argument and do two asserts - the first one that this argument is bigger than 0 and the second - that it is less than 100. and then use this method for 3 coordinates of some random vector

Reply:
"this implementation will verify if y is bigger than 0 (and ignore that it can be more than 100) and that z is just less than 100 (but ignore that it can be less than 0 as well)"  -  it was assumed to check random vectors "randomly" to avoid extra work.
I extended this check a bit.